src/main/java/ch/njol/skript/bukkitutil/block/NewBlockCompat.java

```diff
@@ -315,13 +319,30 @@ private BlockFace findWallTorchSide(Block block) {
 			
 			return null; // Can't place torch here legally
 		}
-		
+
 		@Override
 		public void sendBlockChange(Player player, Location location, Material type, @Nullable BlockValues values) {
 			BlockData blockData = values != null ? ((NewBlockValues) values).data : type.createBlockData();
 			player.sendBlockChange(location, blockData);
 		}
-		
+
+		@Override
+		public void sendBlockChanges(Player player, Location[] locations, Material type, @Nullable BlockValues values, boolean suppressLightUpdates) {
+			BlockData blockData = values != null ? ((NewBlockValues) values).data : type.createBlockData();
```

Comment:
BlockValues won't always be NewBlockValues if you're accepting the BlockValues implementation class. Potentially change this to support BlockCompat too or just have NewBlockValues in the parameter if that's possible in the call references of this method.

This cast cannot be done without checking the instanceof it. ClassCastException is possible here.

Reply:
In reality, it will always be NewBlockValues as the OldBlockValues impl is for 1.9-1.12, of which support has been dropped already. One may ask why is the BlockValues class even a thing nowadays if that's the case and well the reason is simply because refactoring those elements is a huge pain in the ass. Anyhow, I do agree that it should use an instanceof check nonetheless (if there are any forks supporting 1.12, this makes their life easier)